connect/src/com/telenor/connect/ConnectSdk.java

```diff
@@ -136,18 +135,33 @@ private static Intent getAuthIntent(Map<String, String> parameters) {
         final Intent intent = new Intent();
         intent.setClass(getContext(), ConnectActivity.class);
         intent.setAction(ConnectUtils.LOGIN_ACTION);
+        String mccMnc = getMccMnc();
+        WellKnownAPI.WellKnownConfig wellKnownConfig = sdkProfile.getWellKnownConfig();
+        if (!TextUtils.isEmpty(mccMnc) && wellKnownConfig != null &&
+                !(isEmpty(wellKnownConfig.getNetworkAuthenticationTargetIps())
+                        && isEmpty(wellKnownConfig.getNetworkAuthenticationTargetUrls())))
+        {
+            parameters.put("login_hint", String.format("MCCMNC:%s", mccMnc));
+        }
         final String url = getAuthorizeUriAndSetLastAuthState(parameters).toString();
         intent.putExtra(ConnectUtils.LOGIN_AUTH_URI, url);
-        intent.putExtra(ConnectUtils.WELL_KNOWN_CONFIG_EXTRA, sdkProfile.getWellKnownConfig());
+        intent.putExtra(ConnectUtils.WELL_KNOWN_CONFIG_EXTRA, wellKnownConfig);
         return intent;
     }
 
+    private static boolean isEmpty(Collection c) {
```

Comment:
I'm not sure this method belongs here at all.
Use can use CollectionUtils.isEmpty()
but I suggest you to use just wellKnownConfig.getNetworkAuthenticationTargetIps().isEmpty() because you also have implemented returning an emptyset in the getter.